pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -371,17 +373,26 @@ static void validatePartialUpsertStrategies(TableConfig tableConfig, Schema sche
     List<String> primaryKeyColumns = schema.getPrimaryKeyColumns();
     for (Map.Entry<String, UpsertConfig.Strategy> entry : partialUpsertStrategies.entrySet()) {
       String column = entry.getKey();
+      UpsertConfig.Strategy columnStrategy = entry.getValue();
       Preconditions.checkState(!primaryKeyColumns.contains(column), "Merger cannot be applied to primary key columns");
 
       FieldSpec fieldSpec = schema.getFieldSpecFor(column);
       Preconditions.checkState(fieldSpec != null, "Merger cannot be applied to non-existing column: %s", column);
 
-      if (entry.getValue() == UpsertConfig.Strategy.INCREMENT) {
+      if (columnStrategy == UpsertConfig.Strategy.INCREMENT) {
         Preconditions.checkState(fieldSpec.getDataType().getStoredType().isNumeric(),
             "INCREMENT merger cannot be applied to non-numeric column: %s", column);
         Preconditions.checkState(!schema.getDateTimeNames().contains(column),
             "INCREMENT merger cannot be applied to date time column: %s", column);
       }
+
+      if (columnStrategy == UpsertConfig.Strategy.APPEND || columnStrategy == UpsertConfig.Strategy.UNION) {
+        Preconditions
+            .checkState(!fieldSpec.isSingleValueField(), "%s merger cannot be applied to single-value column: %s",
+                columnStrategy.toString(), column);
+        Preconditions.checkState(!schema.getDateTimeNames().contains(column),
```

Comment:
This is redundant as time cannot be MV

Reply:
gotcha, removed this precondition check.